### PR TITLE
	modified:   azure-resources/Sql/servers/kql/74c2491d-048b-0041-a140-…

### PR DESCRIPTION
**Pull request by sethjiusa**
…935960220e20.kql

	modified:   azure-resources/Sql/servers/kql/943c168a-2ec2-a94c-8015-85732a1b4859.kql
	modified:   azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql
	modified:   azure-resources/Sql/servers/kql/d6ef87aa-574e-584e-a955-3e6bb8b5425b.kql
	modified:   azure-resources/Sql/servers/recommendations.yaml

# Overview/Summary

Changed the wording and KQL queries for SQL DB

## Related Issues/Work Items

Replace this with a list of related GitHub Issues and/or ADO Work Items (Internal Only)

- To associate a GitHub Issue, use a [key word](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) preceded with the GitHub issue number.
- To associate an ADO Work Item, use the key word `AB#` succeeded with the [ADO Work Item ID](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## This PR fixes/adds/changes/removes

	modified:   azure-resources/Sql/servers/kql/943c168a-2ec2-a94c-8015-85732a1b4859.kql
	modified:   azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql
	modified:   azure-resources/Sql/servers/kql/d6ef87aa-574e-584e-a955-3e6bb8b5425b.kql
	modified:   azure-resources/Sql/servers/recommendations.yaml

### Breaking Changes

1. _Replace me_
2. _Replace me_

## As part of this pull request I have

- [X] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [X] Ensured PR tests are passing
- [X] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
image
image
image
image
image
image
image

- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql [hunk 1]
```diff
@@ -1,2 +1,2 @@
-// under-development
+// No query availabale. Please refer to resources documentation.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql:1]:
```suggestion
// cannot-be-validated-with-arg
```

File: azure-resources/Sql/servers/kql/d6ef87aa-574e-584e-a955-3e6bb8b5425b.kql [hunk 2]
```diff
@@ -1,2 +1,2 @@
-// under-development
+// No KQL Query available. Please refer to resources documentation.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/kql/d6ef87aa-574e-584e-a955-3e6bb8b5425b.kql:1]:
```suggestion
// cannot-be-validated-with-arg
```

File: azure-resources/Sql/servers/recommendations.yaml [hunk 3]
```diff
@@ -6,9 +6,9 @@
   recommendationResourceType: Microsoft.Sql/servers
   recommendationMetadataState: Active
   longDescription: |
-    If your primary database fails, perform a manual failover to the secondary database which remains read-only until then. Active geo-replication allows creating readable replicas and manual failover in case of a datacenter outage or application upgrade.
+    Active Geo Replication is a business continuity solution that utilizes a readable secondary database replica. In the event of a primary database failure, perform a manual failover to the secondary database and ensure your application points to the appropriate endpoint. Until you fail over, the secondary database remains read-only. Active geo-replication enables you to create readable replicas and manually failover to any replica if there is a datacenter outage. Up to four secondaries are supported in the same or different regions, and the secondaries can also be used for read-only access queries.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/recommendations.yaml:9]: Please re-word to bring the longDescription under the 300 character limit, including spaces.

File: azure-resources/Sql/servers/recommendations.yaml [hunk 4]
```diff
@@ -17,17 +17,17 @@
...
+- description: Use Auto Failover Groups that can include one or multiple databases, typically used by the same application
   aprlGuid: 943c168a-2ec2-a94c-8015-85732a1b4859
   recommendationTypeId: null
   recommendationControl: Disaster Recovery
   recommendationImpact: High
   recommendationResourceType: Microsoft.Sql/servers
   recommendationMetadataState: Active
   longDescription: |
-    You can use the readable secondary databases to offload read-only query workloads. Autofailover groups involve multiple databases configured on a primary server, supporting replication of all databases in the group to only one secondary server or instance in a different region.
+    Failover Groups are a disaster recovery feature that enables you to configure databases contained on a logical server to replicate to a logical server in another region. Failover groups simplify the management of geo-replicated databases and enable a single endpoint that automatically routes connections to the replicated databases in the event of a failure of the primary logical server.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/recommendations.yaml:28]: Please re-word to bring the longDescription under the 300 character limit, including spaces.

File: azure-resources/Sql/servers/recommendations.yaml [hunk 5]
```diff
@@ -46,9 +46,9 @@
   recommendationResourceType: Microsoft.Sql/servers
   recommendationMetadataState: Active
   longDescription: |
-    By default, the premium availability model clusters nodes in one datacenter.
+    By default, when you provision an Azure SQL Database with the premium tier, multiple copies of the database are created within the same azure region. To support geo redundancy, Databases can be provisioned (or existing databases can be modified) as Zone Redundant, which locates the copies in different Azure Availability Zones. This will ensure database availability in the event of a regional outage.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/recommendations.yaml:49]: Please re-word to bring the longDescription under the 300 character limit, including spaces.

File: azure-resources/Sql/servers/recommendations.yaml [hunk 6]
```diff
@@ -17,17 +17,17 @@
     - name: Active Geo Replication
       url: "https://learn.microsoft.com/en-us/azure/azure-sql/database/active-geo-replication-overview"
 
-- description: Auto Failover Groups for apps should include all related databases for cohesion
+- description: Use Auto Failover Groups that can include one or multiple databases, typically used by the same application
```
Comment (ejhenry) [inline azure-resources/Sql/servers/recommendations.yaml:20]: Please re-word to bring the description under the 100 character limit, including spaces.

File: azure-resources/Sql/servers/recommendations.yaml [hunk 7]
```diff
@@ -65,9 +65,9 @@
   recommendationResourceType: Microsoft.Sql/servers
   recommendationMetadataState: Active
   longDescription: |
-    Azure SQL Database, known for its resilience to infrastructure failures, can occasionally encounter connectivity issues due to transient errors. Implementing retry logic in your code ensures continued operation by reattempting failed calls, maintaining smooth database interaction.
+    When transient failures occur between the application and the Azure SQL Database environment, it is important that the application manages connection retries in an appropriate manner. There is no configuration necessary at the Database layer, but the application must be configured to retry gracefully.
```
Comment (ejhenry) [inline azure-resources/Sql/servers/recommendations.yaml:68]: Please re-word to bring the longDescription under the 300 character limit, including spaces.

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @sethjiusa! Just a few items to address to bring some of the content under the character limits.

For the updated ARG query, please rename the GeoReplicationState column to param1 and include in the PR a screenshot of the successful testing of the query.

File: azure-resources/Sql/servers/recommendations.yaml (hunk 6 above)
Comment (sethjiusa) [reply azure-resources/Sql/servers/recommendations.yaml:20]: This line is 88 characters including spaces. Not sure why this is above 100 characters?

File: azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql (hunk 1 above)
Comment (sethjiusa) [reply azure-resources/Sql/servers/kql/cbb17a29-64fb-c943-95d0-8df814a37c40.kql:1]: Fixed

File: azure-resources/Sql/servers/recommendations.yaml (hunk 3 above)
Comment (sethjiusa) [reply azure-resources/Sql/servers/recommendations.yaml]: Fixed

File: azure-resources/Sql/servers/recommendations.yaml (hunk 4 above)
Comment (sethjiusa) [reply azure-resources/Sql/servers/recommendations.yaml:28]: Fixed

**Comment by sethjiusa** [comment]
> Thanks for the contribution @sethjiusa! Just a few items to address to bring some of the content under the character limits.
> 
> For the updated ARG query, please rename the GeoReplicationState column to param1 and include in the PR a screenshot of the successful testing of the query.

-- Sunil Seth -- Fixed Here is the screenshot of the query along with the result
image

File: azure-resources/Sql/servers/recommendations.yaml (hunk 6 above)
Comment (ejhenry) [reply azure-resources/Sql/servers/recommendations.yaml:20]: Word and a couple online character counters show this as 107 characters, including spaces.

File: azure-resources/Sql/servers/recommendations.yaml (hunk 7 above)
Comment (ejhenry) [reply azure-resources/Sql/servers/recommendations.yaml:68]: My count returns 302 characters, including spaces.

File: azure-resources/Sql/servers/kql/74c2491d-048b-0041-a140-935960220e20.kql [hunk 8]
```diff
@@ -1,7 +1,6 @@
...
-// Provides a list of SQL databases that are not configured for Geo-redundant storage.
-resources
+// Provides a list of SQL databases that are not part of Geo Replication.
+Resources
 | where type == "microsoft.sql/servers/databases"
-| where  (properties['currentBackupStorageRedundancy'] ) <> 'Geo'
-| project recommendationId = "74c2491d-048b-0041-a140-935960220e20", name, id, tags, param1=strcat("CurrentGeoRedudancy=", properties['currentBackupStorageRedundancy'] )
-
+| extend param1 = iif(isnotempty(properties.geoReplicationRole), "Part of Geo Replication", "Not part of Geo Replication")
+| project recommendationId = "74c2491d-048b-0041-a140-935960220e20", DatabaseName = name, id, tags, param1
```
Comment (ejhenry) [inline azure-resources/Sql/servers/kql/74c2491d-048b-0041-a140-935960220e20.kql:6]:
```suggestion
| project recommendationId = "74c2491d-048b-0041-a140-935960220e20", DatabaseName = name, id, tags, param1

```

**Review by ejhenry — APPROVED** [review]
LGTM

**Comment by Dost2010** [comment]
Hello @Sethjimsft
Can you please re-name column DatabaseName to name only

**Comment by sethjiusa** [comment]
@Dost2010 renamed the databasename column to name.

**Review by Dost2010 — APPROVED** [review]

**Pull request merged**